pages/models/generations/Sources/VSC/EMTGridFollowingVSC/index.md

```diff
@@ -0,0 +1,496 @@
+</div>
+<br>
+
+To determine the PI controller parameters, we propose the complete power loop computing the error response, including the current loop dynamics with the obtained transfer function. The following block diagram shows the complete power loop:
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+<img src="{{ '/pages/models/generations/Sources/VSC/EMTGridFollowingVSC/ErrorPowerLoopVSC.svg' | relative_url }}"
+     alt="Power Control Diagram"
+     style="float: center; margin-right: 10px; height: 280px;" />
+</div>
+<div align = 'center'>
+Figure 7: Closed-Loop Power Control Diagram
+</div>
+<br>
+
+Following a similar procedure to the one of the current loop, the closed-loop transfer function of the power loop can be modelled as a first-order system:
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$ \frac{P(s)}{P^*(s)} = \frac{Q(s)}{Q^*(s)} = \frac{1}{\tau_p s + 1} $$
+</div>
+
+where $$\tau_p$$ is the time constant of the power loop. The PI controller can be obtained with the use of the time constant of the current loop as:
+
+<div style="background-color:rgba(0, 0, 0, 0.0470588); text-align:center; vertical-align: middle; padding:4px 0;">
+
+$$K^{ipl}_p = \frac{2\tau_c}{3V_{peak}\tau_p}$$ 
+$$K^{ipl}_i = \frac{2}{3V_{peak}\tau_p}$$
+</div>
+
+The time constant $$\tau_p$$ will be larger than $$\tau_c$$, since the power loop is designed to have a slower response than the current loop.
+
+### Limitations of current
+
+The technical constraints of the VSC can be included in the controls using saturation blocks. Depending on the desired operation mode, the converter can be set to prioritize one of the current components. These operation modes are typically defined by the grid codes, although a possible implementation could be the following:
+
+* **Normal operation**: The converter will prioritize the $$i^q$$ component, then $$i^q_{max} = I_{max}$$ and $$i^d_{max} = \sqrt{I_{max}^2 - \max{i^q, i^{q*}}} $$.
+* **Transient or fault operation**: The converter will prioritize the $$i^d$$ component, then $$i^d_{max} = I_{max}$$ and $$i^q_{max} = \sqrt{I_{max}^2 - \max{i^d, i^{d*}}^2} $$.
+
+
+### Modulation
+
+As explained in the current loop section, the converter voltage setpoint obtained is considered directly the converter voltage. However, this sinusoidal voltage is obtained from a DC source, and to generate the AC voltage, the IGBT switching is modulated using Pulse-Width Modulation (PWM). To determine the position of the IGBTs, the Space Vector PWM (SVPWM) technique is used [[9]](#9). It is based on the projection of the AC side converter voltages in a space vector. Considering the possible configurations of the IGBTs, there are a total of $$2^3 = 8$$ possible states, the resulting voltages of which can be represented in a hexagon in the $$\alpha\beta$$ reference frame:
```

Comment:
The PWM in this case is done using this modulation method called SVPWM, instead of using carrier waves. The output of this modulation will be the On/Off position of the IGBTs in order to generate the desired voltage.
There are 8 positions, two of which are in the center of the diagram (positions 0,7), as the voltage output of this swithing state is 0.
I will rewrite the whole section to follow a better order